code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -34,124 +31,318 @@
+	to_chat(predatoralien, SPAN_XENOWARNING("We feel our muscles tense as our speed and armor increase!"))
+	speed_buff = TRUE
+	armor_buff = TRUE
+	predatoralien.speed_modifier -= speed_buff_amount
+	predatoralien.armor_modifier += armor_buff_amount
+	predatoralien.recalculate_speed()
+	predatoralien.recalculate_armor()
+	playsound(predatoralien, 'sound/voice/predalien_growl.ogg', 75, 0, status = 0)
+	apply_cooldown()
 
-			addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(unroot_human), carbon, TRAIT_SOURCE_ABILITY("Smash")), get_xeno_stun_duration(carbon, freeze_duration))
 
+/datum/action/xeno_action/onclick/feralrush/proc/remove_rush_effects()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/predatoralien = owner
+	if (speed_buff == TRUE)
+		to_chat(predatoralien, SPAN_XENOWARNING("Our muscles relax as we feel our speed wane."))
+		predatoralien.remove_filter("predalien_toughen")
+		predatoralien.speed_modifier += speed_buff_amount
+		predatoralien.recalculate_speed()
+		speed_buff = FALSE
+		remove_rush_effects()
 
-	for(var/mob/M in view(xeno))
-		if(M && M.client)
-			shake_camera(M, 0.2 SECONDS, 1)
+/datum/action/xeno_action/onclick/feralrush/proc/remove_armor_effects()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/predatoralien = owner
+	if (armor_buff == TRUE)
+		to_chat(predatoralien, SPAN_XENOWARNING("We are no longer armored."))
+		predatoralien.armor_modifier -= armor_buff_amount
+		predatoralien.recalculate_armor()
+		armor_buff = FALSE
+		remove_armor_effects()
 
-	apply_cooldown()
-	return ..()
 
-/datum/action/xeno_action/activable/devastate/use_ability(atom/target)
+/datum/action/xeno_action/onclick/toggle_gut_targetting/use_ability(atom/A)
+
 	var/mob/living/carbon/xenomorph/xeno = owner
+	var/action_icon_result
 
-	if (!action_cooldown_check())
+	if(!xeno.check_state())
 		return
 
-	if (!xeno.check_state())
+	var/datum/action/xeno_action/activable/feralfrenzy/guttype = get_xeno_action_by_type(xeno, /datum/action/xeno_action/activable/feralfrenzy)
+	if (!guttype)
 		return
 
-	if (!isxeno_human(target) || xeno.can_not_harm(target))
-		to_chat(xeno, SPAN_XENOWARNING("We must target a hostile!"))
+	if (guttype.targetting == SINGLETARGETGUT)
+		action_icon_result = "rav_scissor_cut"
+		guttype.targetting = AOETARGETGUT
+		to_chat(xeno, SPAN_XENOWARNING("We will now attack everyone around us."))
+	else
+		action_icon_result = "gut"
+		guttype.targetting = SINGLETARGETGUT
+		to_chat(xeno, SPAN_XENOWARNING("We will now focus our rage on one person!"))
+
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions_xeno.dmi', button, action_icon_result)
+	return ..()
+
+
+
+
+
+
+/datum/action/xeno_action/activable/tail_stab/predalien_tailstab/use_ability(atom/targetted_atom)
+	var/mob/living/carbon/xenomorph/predalien_tailstab = owner
+
+
+	if(!action_cooldown_check())
+		return FALSE
+
+	if(!predalien_tailstab.check_state())
+		return FALSE
+
+	if (world.time <= predalien_tailstab.next_move)
+		return FALSE
```

Comment:
I don't think we need to return FALSE since the returned value of use_ability() isn't checked anywhere I know of.